hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CombinedBlockCache.java

```diff
@@ -79,7 +79,13 @@ public Cacheable getBlock(BlockCacheKey cacheKey, boolean caching,
       boolean repeat, boolean updateCacheMetrics) {
     // TODO: is there a hole here, or just awkwardness since in the lruCache getBlock
     // we end up calling l2Cache.getBlock.
-    return lruCache.containsBlock(cacheKey)?
+    boolean existInL1 = lruCache.containsBlock(cacheKey);
+    if (!existInL1 && updateCacheMetrics) {
```

Comment:
Within the LRUBlock cache we do stats.miss when its repeat = false
if (!repeat && updateCacheMetrics) {
        stats.miss(caching, cacheKey.isPrimary(), cacheKey.getBlockType());
      }
Checked 2.x based my version and I believe in 1.x also same.  Pls check and adjust. 
Else patch looks good. And simple. :-)
Very very nice catch (again)

Reply:
If existInL1 is false, we will not check LRU cache here so that the miss will not be double counted.

Reply:
Yes, we should care about the repeat situation.